Ptref: add a period filter on /disruptions

antoine-de opened the pull request:
now since and until filter in the impacts application_periods

you can thus do `api.navitia.io/disruptions?since=20160205T1615`

to get all disruptions that have an application period after the since

we return the disruptions that have a non null intersection between one of ther application period and the [since, until] period

antoine-de committed:
Ptref: add a period filter on /disruptions

now since and until filter in the impacts application_periods

xlqian commented on source/ptreferential/ptreferential.cpp:
```diff
@@ -394,10 +394,45 @@ static bool keep_vj(const nt::VehicleJourney* vj,
+
+    std::vector<idx_t> res;
+    for (const idx_t idx: indexes) {
+        auto impact = data.pt_data->disruption_holder.get_weak_impacts()[idx].lock();
```
shouldn't impact be tested before being de-referenced?

antoine-de replied:
yes you're right, done

antoine-de committed:
correction after review

antoine-de committed:
ptref: the end of the filtering period is now included

since we want since=now&until=now to be non empty the filtering periods
are [since, until]
